--- Settings.java
+++ Settings.java
@@ -667,6 +667,8 @@ public class Settings
 
 	static void setEnvVars(MainActivity p)
 	{
+		nativeSetEnv( "AUDIO_NATIVE_SAMPLE_RATE", String.valueOf(android.media.AudioTrack.getNativeOutputSampleRate(android.media.AudioManager.STREAM_MUSIC)) );
+		nativeSetEnv( "OS_VERSION", System.getProperty("os.version", "4.0.0") );
 		String lang = new String(Locale.getDefault().getLanguage());
 		if( Locale.getDefault().getCountry().length() > 0 )
 			lang = lang + "_" + Locale.getDefault().getCountry();
--- Video.java
+++ Video.java
@@ -1259,7 +1259,7 @@ class DemoGLSurfaceView extends GLSurfaceView_SDL {
 				Globals.TouchscreenKeyboardTheme = 9;
 
 			if( Globals.TouchscreenKeyboardTheme == 0 )
-				Settings.nativeSetupScreenKeyboardButtons(loadRaw(p, R.raw.ultimatedroid));
+				Settings.nativeSetupScreenKeyboardButtons(loadRaw(p, R.raw.xsdl));
 			if( Globals.TouchscreenKeyboardTheme == 1 )
 				Settings.nativeSetupScreenKeyboardButtons(loadRaw(p, R.raw.simpletheme));
 			if( Globals.TouchscreenKeyboardTheme == 2 )
